app/assets/javascripts/discourse/controllers/composer.js.es6

```diff
@@ -770,23 +776,30 @@ export default Ember.Controller.extend({
           .then(resolve, reject);
       }
 
-      // we need a draft sequence for the composer to work
-      if (opts.draftSequence === undefined) {
-        return Draft.get(opts.draftKey)
-          .then(function(data) {
-            opts.draftSequence = data.draft_sequence;
-            opts.draft = data.draft;
-            self._setModel(composerModel, opts);
-          })
-          .then(resolve, reject);
-      }
-
       if (composerModel) {
         if (composerModel.get("action") !== opts.action) {
           composerModel.setProperties({ unlistTopic: false, whisper: false });
         }
       }
 
+      // check if there is another draft saved on server
+      // or get a draft sequence number
+      if (!opts.draft || opts.draftSequence === undefined) {
+        return Draft.get(opts.draftKey)
+          .then(data => self.confirmDraftAbandon(data))
+          .then(data => {
+            // we need a draft sequence for the composer to work
```

Comment:
Since this is encapsulated inside the `if (!opts.draft || opts.draftSequence === undefined) {`, we can simply this to

```javascript
opts.draft = opts.draft || data.draft;
opts.draftSequence = data.draft_sequence;
self._setModel(composerModel, opts);
```